Added Detach/Attach functions. Updated FreeProcess docs

thomaspreece opened the pull request:
Added Detach/Attach functions to FreeProcess so you can run processes that don't terminate when you close your app. Also updated the FreeProcess documentation along with examples. 

Feel free to reject the Detach/Attach functions but I personally tend to use them quite a bit with the software I write.

GWRon commented:
if this is of interest...you might add it to maxmods too..as it does not "need" bmx-ng and should work with vanilla bmx too

woollybah commented:
Looks fine, except that I think "DetachProcess" sounds better than "ProcessDetach", for action-type functions.
ProcessStatus is different because it is a request for information.

thomaspreece commented:
Ok, I changed the examples + code to use DetachProcess/ AttachProcess.
Not entirely sure why I did it the other way but I changed this code many many months ago so I can't remember now

woollybah commented:
I thought you closed it by mistake...

thomaspreece commented:
Sorry I'm want to make a few changes to the documentation first before you merge it (hence why I closed it)

GWRon commented:
Hint: for new enhancements you should think of using "branches" - so you can keep your fork clean of "invalid" commits. Push the branch as pull request and upon "accept" you just remove your branch and pull the changes from "main" to your fork.

thomaspreece commented:
Can you reopen this, it appears I have lost the permissions to do so as you closed it last
